src/core/qgsfield.cpp

```diff
@@ -258,6 +259,15 @@ QString QgsField::displayString( const QVariant &v ) const
     return QgsApplication::nullRepresentation();
   }
 
+  if ( v.userType() == QMetaType::type( "QgsReferencedGeometry" ) )
+  {
+      QgsReferencedGeometry geom = qvariant_cast<QgsReferencedGeometry>( v );
+      if( geom.isNull() )
+          return QgsApplication::nullRepresentation();
+      else
+          return QStringLiteral( "%1 [%2]" ).arg( geom.asWkt(), geom.crs().userFriendlyIdentifier() );
```

Comment:
I wonder here if we should truncate the wkt after say 1000 characters. I've seen the attribute table ground to a halt before when using secondary geometry columns because of massive WKT text strings